File: model/src/main/java/ru/mipt/acsl/decode/modeling/TransformationResult.java
package ru.mipt.acsl.decode.modeling;

import org.jetbrains.annotations.NotNull;

import java.util.List;
import java.util.Optional;

/**
 * @author Artem Shein
 */
public interface TransformationResult<T>
{
    @NotNull
    Optional<T> getResult();
    @NotNull
    List<TransformationMessage> getMessages();
    boolean hasError();
}

File: model/src/main/scala/ru/mipt/acsl/modeling/TransformationResult.scala
package ru.mipt.acsl.modeling

/**
  * @author Artem Shein
  */
trait TransformationResult[T] {
  def result: Option[T]
  def messages: Seq[TransformationMessage]
  def hasError: Boolean = messages.exists(_.level == ErrorLevel)
}
